pkg/manager/repo/validator.go

```diff
@@ -0,0 +1,111 @@
+package repo
+
+
+import (
+	"encoding/json"
+	"fmt"
+	"errors"
+	"strings"
+	"net/http"
+	"regexp"
+	neturl "net/url"
+	qs_conf "github.com/yunify/qingstor-sdk-go/config"
+	qs "github.com/yunify/qingstor-sdk-go/service"
+	"openpitrix.io/openpitrix/pkg/logger"
+)
+
+
+var(
+	compRegEx = regexp.MustCompile(`^(?P<bucket>.+)\.(?P<zone>.+)\.qingstor\.com$`)
+)
+
+
+func validateVisibility(visibility string) error {
+	switch visibility {
+	case "public":
+	case "private":
+	case "labeled":
```

Comment:
What is the meaning of `labeled`?

Reply:
When a repo is labeled, user can access it if his repo-selector has a label match it.